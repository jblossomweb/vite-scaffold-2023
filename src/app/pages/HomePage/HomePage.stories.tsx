import React from 'react'
import { MemoryRouter } from 'react-router'
import type { Meta, StoryObj } from '@storybook/react'

import HomePage from './HomePage'

const meta = {
  title: 'pages/HomePage',
  component: HomePage,
  parameters: {
    layout: 'fullscreen',
  },
  render: () => (
    <MemoryRouter>
      <HomePage />
    </MemoryRouter>
  ),
} satisfies Meta<typeof HomePage>

export default meta
type Story = StoryObj<typeof meta>

export const Sample: Story = {
  args: {},
}
