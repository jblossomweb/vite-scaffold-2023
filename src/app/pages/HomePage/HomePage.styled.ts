import styled from 'styled-components'
import tw from 'tailwind-styled-components'

export const Logo = styled.img`
  height: 40vmin;
  pointer-events: none;
`

export const LogoSpin = tw(Logo)`
  animate-spin
`
