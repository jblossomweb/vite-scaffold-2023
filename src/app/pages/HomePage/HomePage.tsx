import React from 'react'
import logo from 'svgs/logo.svg'
import HeroTemplate from 'app/templates/HeroTemplate'

import * as Styled from './HomePage.styled'

const HomePage: React.FC = () => (
  <HeroTemplate title="Home Page">
    <Styled.LogoSpin
      src={logo}
      alt="logo"
    />
    <p>
      Edit <code>src/app/pages/HomePage</code> and save to reload.
    </p>
    <a
      href="https://reactjs.org"
      target="_blank"
      rel="noopener noreferrer"
    >
      Learn React
    </a>
  </HeroTemplate>
)

export default HomePage
