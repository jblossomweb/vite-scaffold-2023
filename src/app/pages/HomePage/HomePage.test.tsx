import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render, screen } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import HomePage from './HomePage'
import * as stories from './HomePage.stories'

describe('app/pages/HomePage', () => {
  describe('stories', () => {
    storySnaps(stories)
  })

  describe('assertions', () => {
    const renderScreen = () =>
      render(
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>,
      )

    it('renders a logo', () => {
      renderScreen()
      const logo = screen.getByAltText(/logo/i)
      expect(logo).toBeInTheDocument()
    })

    it('renders a spinning logo', () => {
      renderScreen()
      const logo = screen.getByAltText(/logo/i)
      expect(logo).toHaveClass('animate-spin')
    })

    it('renders learn react link', () => {
      renderScreen()
      const linkElement = screen.getByText(/learn react/i)
      expect(linkElement).toBeInTheDocument()
    })
  })
})
