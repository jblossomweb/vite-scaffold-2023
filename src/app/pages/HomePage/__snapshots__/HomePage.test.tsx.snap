// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/pages/HomePage stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <header
      class="min-h-screen flex flex-col items-center justify-center"
    >
      <img
        alt="logo"
        class="sc-bdnyFh dokdYg animate-spin"
      />
      <p>
        Edit 
        <code>
          src/app/pages/HomePage
        </code>
         and save to reload.
      </p>
      <a
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</DocumentFragment>
`;
